src/harvesters/SaddlePoolManager.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import "BoringSolidity/interfaces/IERC20.sol";
+
+import "../external/saddle/ISwap.sol";
+
+contract SaddlePoolManager {
+    function addLiquidity(
+        address swap,
+        uint256 amountIn,
+        uint256 minToMint
+    ) public returns (uint256 LPTokens) {
+        ISwap hopSwap = ISwap(swap);
+        IERC20 token0 = IERC20(hopSwap.getToken(0));
+        uint256 tokensIn = amountIn;
+
+        if (amountIn == 0) {
```

Comment:
is there any reason you are doing that? I think you should not pass `amountIn` as a param and use the balance

Reply:
I'm doing this for calculation sake. When vault-harvester use this as a view function the vault doesn't have this token yet